packages/react-draggable-dialog/src/components/DraggableDialog/useDraggableDialog.ts

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { useFluent, useEventCallback } from '@fluentui/react-components';
+
+function roundByDPR(value: number, document?: Document) {
+  const win = document?.defaultView;
+
+  if (!win) {
+    return value;
+  }
+
+  const dpr = win.devicePixelRatio || 1;
+
+  return Math.round(value * dpr) / dpr;
+}
+
+type DraggableMarginAxis = {
+  mainAxis?: number;
+  crossAxis?: number;
+};
+type DraggableMarginViewport = {
+  top?: number;
+  end?: number;
+  bottom?: number;
+  start?: number;
+};
+type DraggableDialogOptions = {
+  keepInViewport?: boolean;
+  margin?: number | DraggableMarginAxis | DraggableMarginViewport;
+};
+
+const getDraggableMargin = (
+  margin: DraggableDialogOptions['margin']
+): Required<DraggableMarginViewport> => {
+  const defaultMargin = {
+    top: 0,
+    end: 0,
+    bottom: 0,
+    start: 0,
+  };
+
+  if (!margin) {
+    return defaultMargin;
+  }
+
+  if (typeof margin === 'number') {
+    return {
+      top: margin,
+      end: margin,
+      bottom: margin,
+      start: margin,
+    };
+  }
+
+  if ('mainAxis' in margin || 'crossAxis' in margin) {
+    const x = margin.mainAxis || 0;
+    const y = margin.crossAxis || 0;
+
+    return {
+      top: y,
+      end: x,
+      bottom: y,
+      start: x,
+    };
+  }
+
+  return {
+    ...defaultMargin,
+    ...margin,
+  };
+};
+
+export const useDraggableDialog = (_opts: DraggableDialogOptions) => {
```

Comment:
what's the deal with `_` prefix ?

Reply:
This file was included by mistake, as it used to contain logic that was separated and placed in the right places. I removed it.